app/src/main/java/com/jxc/app/controllers/UserController.java

```diff
@@ -28,15 +29,29 @@
     }
 
     /**
-     * Method to find a user by id
-     * @param id The id of the {@link User}
-     * @return A {@link ResponseEntity} that contains the {@link User} if he exist, throw an {@link UserNotFoundException} otherwise
-     * @throws UserNotFoundException An exception with a message that display the id of the non existing {@link User}
+     * Method to find a user by email
+     * @param userId The email of the {@link User}
+     * @return A {@link ResponseEntity} that contains the {@link UserInfosDTO} if he exist, throw an {@link UserNotFoundException} otherwise
+     * @throws UserNotFoundException An exception with a message that display the email of the non existing {@link User}
      */
-    @GetMapping("/{id}")
-    public ResponseEntity<User> getUserById(@PathVariable String id) throws UserNotFoundException {
-            User user = this.userService.findUserById(id);
-            return new ResponseEntity<>(user, HttpStatus.OK);
+    @GetMapping("/{userId}")
+    public ResponseEntity<UserInfosDTO> getUserByEmail(@PathVariable String userId) throws UserNotFoundException {
+            User user = this.userService.findUserByEmail(userId);
+            UserInfosDTO userInfo = this.userService.UserToUserInfosDTO(user);
```

Comment:
ça fonctionne, mais quitte à convertir comme ça, on préfèrera mettre la méthode dans la classe user.
Si tu veux faire ça plus proprement et apprendre de nouvelles choses (pas complexes), fait plutôt un mapper : 
https://www.baeldung.com/entity-to-and-from-dto-for-a-java-spring-application
https://www.baeldung.com/dozer